### ISSUE TITLE
Discuss: Fork Helix into Pinot Repo

### ISSUE THREAD

**Issue by kishoreg** [apache/incubator-pinot#4982]
I never thought I would be the one initiating this thread. But over the last year, the quality of Helix code has gone down drastically and it's preventing us to make a successful pinot release. Many changes are not well thought out and are being designed to meet the needs of specific use cases. Helix is an important component for Pinot and its important for us to have control over its behavior. 

My proposal is to fork the helix module into Pinot. Please let me know if anyone has any concerns.

**Comment by lei-xia** [comment]
Kishore, we are working with Pinot team on the testing and I think we are close to get a stable release for Pinot.  We also have on-going discussions with Pinot to smooth upgrade process. I would like to hear more of your suggestions on how we can improve, but in my opinion, I do not think having a fork of Helix code in Pinot can solve the problem.

**Comment by kishoreg** [comment]
I don’t think this is just a process problem, though it definitely adds to the problem. We will keep the interface as it is and we might try to migrate later in future once Helix becomes stable. 
As of now, setup is counter productive

**Comment by lei-xia** [comment]
The new Apache 0.9.4 will be released this week for Pinot. We have tested this release with Pinot thoroughly within LinkedIn's internal prod deployment, it should be stable for Pinot to upgrade its open-source dependancy.

**Comment by xiangfu0** [comment]
Closing — no longer relevant. The decision was made to continue using the official Apache Helix release (currently v1.3.2) rather than forking it into the Pinot repo.
